File: 2021/April/Day 28/detect-capital.py
# source : https://leetcode.com/problems/detect-capital/submissions/
class Solution:
    def detectCapitalUse(self, word: str) -> bool:

        capcount =0;

        firstCap = ord(word[0])>64 and ord(word[0])<97
        if(firstCap):
            capcount =1;
        for i in range(1,len(word)):

            if ord(word[i])>64 and ord(word[i])<97:
                capcount = capcount = capcount+1

        if(capcount == len(word)) or (capcount==0) or (capcount==1 and firstCap):
            return True
        else:
            return False

File: 2022/June /Day 16/detectCapital.cpp
class Solution {
public:

    int isCap(char c)
    {

        if(int(c)>63 && int(c)<97)
        {
            return 1;
        }
        return 0;
    }
    bool detectCapitalUse(string word)
    {

       vector<int> out;
        for(auto&i:word)
        {
            out.insert(out.end(),isCap(i));
        }

        int f=out[0];
        if(size(word)<2){return true;}
        int s=out[1];
        for(int i=1;i<size(word);i++)
        {
            if(f==0)
            {
                if(out[i]==1){
                    // cout<<"I is "<<i<<out[i];
                    // cout<<"Returned Here";
                    return false;}

            }


            else if(s==1)
            {
                if(out[i]!=1){return false;}
            }

            else if(s==0)
            {
                if(out[i]==1){return false;}
            }


        }
        return true;
    }
};
